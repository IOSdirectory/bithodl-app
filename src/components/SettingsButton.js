import React, { Component } from 'react';
import { StyleSheet, View, TouchableHighlight, ActivityIndicator } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  container: {
    paddingTop: 12,
    paddingBottom: 12,
    paddingRight: 15,
    marginLeft: 15,
    borderBottomColor: '#DDDDDF',
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  label: {
    fontWeight: '500',
    color: '#D49146'
  },
  disabled: {
    color: '#8F8E94'
  },
  loader: {
    position: 'absolute',
    top: 1,
    right: 15,
    height: 42
  }
});

export default class SettingsButton extends Component {
  _onPress() {
    if (!this.props.loading) {
      this.props.onPress();
    }
  }

  render() {
    const { isLastItem, loading } = this.props;
    const title = loading ? this.props.loadingTitle : this.props.title;
    const underlayColor = loading ? 'white' : '#FAFAFA';

    const containerStyles = [
      styles.container,
      isLastItem ? { borderBottomWidth: 0 } : undefined
    ];

    const labelStyles = [
      styles.label,
      loading ? styles.disabled : undefined
    ];

    return (
      <TouchableHighlight onPress={this._onPress.bind(this)} underlayColor={underlayColor}>
        <View style={containerStyles}>
          <StyledText style={labelStyles}>{title}</StyledText>
          { loading ? <ActivityIndicator animating={true} color='#8F8E94' style={styles.loader} size='small' /> : null }
        </View>
      </TouchableHighlight>
    );
  }
}

SettingsButton.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  isLastItem: PropTypes.bool,
  loading: PropTypes.bool,
  loadingTitle: PropTypes.string
};
