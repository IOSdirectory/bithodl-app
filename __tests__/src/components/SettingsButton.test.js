import React from 'react';
import renderer from 'react-test-renderer';
import SettingsButton from '../../../src/components/SettingsButton';

describe('SettingsButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsButton
        title='dbf83264-0385-43a8-9cbb-ff52c1587401'
        onPress={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as the last item', () => {
    const tree = renderer.create(
      <SettingsButton
        title='8923b772-b0b3-4cb9-9ec1-f08516a80dd5'
        onPress={() => {}}
        isLastItem={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly while loading', () => {
    const tree = renderer.create(
      <SettingsButton
        title='8923b772-b0b3-4cb9-9ec1-f08516a80dd5'
        onPress={() => {}}
        loading={true}
        loadingTitle='85361618-e8d1-4284-9f2c-928cf417c323'
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
