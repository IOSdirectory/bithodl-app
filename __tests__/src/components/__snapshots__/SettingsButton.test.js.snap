// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#D49146",
              "fontWeight": "500",
            },
            undefined,
          ],
        ]
      }
    >
      dbf83264-0385-43a8-9cbb-ff52c1587401
    </Text>
  </View>
</View>
`;

exports[`SettingsButton renders correctly as the last item 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        Object {
          "borderBottomWidth": 0,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#D49146",
              "fontWeight": "500",
            },
            undefined,
          ],
        ]
      }
    >
      8923b772-b0b3-4cb9-9ec1-f08516a80dd5
    </Text>
  </View>
</View>
`;

exports[`SettingsButton renders correctly while loading 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#D49146",
              "fontWeight": "500",
            },
            Object {
              "color": "#8F8E94",
            },
          ],
        ]
      }
    >
      85361618-e8d1-4284-9f2c-928cf417c323
    </Text>
    <ActivityIndicator
      animating={true}
      color="#8F8E94"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "height": 42,
          "position": "absolute",
          "right": 15,
          "top": 1,
        }
      }
    />
  </View>
</View>
`;
